### PR TITLE
Replace to shakapacker from webpacker

### PR DESCRIPTION
**Pull request by ykzts**
Webpacker is no longer maintained and will be replaced by Shakapacker, a fork of Webpacker.

### CODE REVIEW

File: config/webpack/environments/production.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    //     '**/*.woff',
+    //   ],
+    //   ServiceWorker: {
+    //     entry: `imports-loader?additionalCode=${encodeURIComponent(`var ATTACHMENT_HOST=${JSON.stringify(attachmentHost)};`)}!${encodeURI(join(__dirname, '../../app/javascript/mastodon/service_worker/entry.js'))}`,
+    //     cacheName: 'mastodon',
+    //     output: '../assets/sw.js',
+    //     publicPath: '/sw.js',
+    //     minify: true,
+    //   },
+    // }),
```
Comment (ykzts) [inline config/webpack/environments/production.js:72]: offline-plugin is not compatible with webpack 5. We need to replace it with workbox-webpack-plugin which has similar functionality.

**Comment by ClearlyClaire** [comment]
The things that worried me the most when I had a look at shakapacker were:
> Ensure that your layouts and views will only have **at most one call** to `javascript_pack_tag` and **at most one call** to `stylesheet_pack_tag`

and

> Ensure no nested directories in your `source_entry_path`.

I'm not sure exactly what the latter means for us, but the first one seems like a massive pain for Mastodon and downstreams.

**Comment by ykzts** [comment]
Mastodon overrides the `*_pack_tag`, so multiple calls can be made as before.

Shakapacker can no longer create a directory under `source_entry_path` to place entry scripts, such as `app/javascript/packs/path/to/abs.js` in the previous configuration. Mastodon is not affected because there are no directories under `app/javascript/packs`.

**Comment by ykzts** [comment]
#18409 must be merged first.

**Comment by shleeable** [comment]
#18409 is merged.

**Comment by ThisIsMissEm** [comment]
I'm hearing there's a plan to move to Vite for builds, MAS-80, so perhaps this PR is no longer necessary? cc @renchap I know you've been working on this, if memory serves.

**Comment by renchap** [comment]
I am working on a switch from a Webpack-based toolchain to ViteJS: #24981

It is not 100% working yet but it should provide a much faster and simpler alternative.

**Pull request closed without merging**